framework/src/problems/FEProblemBase.C

```diff
@@ -746,6 +747,53 @@ FEProblemBase::initialSetup()
     }
   }
 
+  // Set up stateful material property redistribution, if we suspect
+  // it may be necessary later.
+#ifdef LIBMESH_ENABLE_AMR
+  if ((_adaptivity.isOn() || _num_grid_steps) &&
+      (_material_props.hasStatefulProperties() || _bnd_material_props.hasStatefulProperties() ||
+       _neighbor_material_props.hasStatefulProperties()))
+  {
+    // Even on a serialized Mesh, we don't keep our material
+    // properties serialized, so we'll rely on the callback to
+    // redistribute() to redistribute properties at the same time
+    // libMesh is redistributing elements.
+    auto add_redistributer =
```

Comment:
I would probably vote to make this a class method as `initialSetup` is already very long